Content.Client/_White/ItemSlotRenderer/ItemSlotRendererSystem.cs

```diff
@@ -0,0 +1,144 @@
+using Content.Client.Hands;
+using Content.Shared.Containers.ItemSlots;
+using Microsoft.CodeAnalysis.Elfie.Serialization;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Input;
+using Robust.Shared.Containers;
+using Robust.Shared.Enums;
+using Robust.Shared.Graphics;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Reflection;
+using Robust.Shared.Timing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Content.Client._White.ItemSlotRenderer;
+
+/// <summary>
+/// I can feel my grip on reality slowly slipping.
+/// </summary>
+public sealed class ItemSlotRendererSystem : EntitySystem
+{
+    [Dependency] private readonly IReflectionManager _reflection = default!;
+    [Dependency] private readonly ItemSlotsSystem _slot = default!;
+    [Dependency] private readonly IClyde _clyde = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ItemSlotRendererComponent, ComponentStartup>(OnStartup);
+        SubscribeLocalEvent<ItemSlotRendererComponent, ComponentRemove>(OnRemove);
+        SubscribeLocalEvent<ItemSlotRendererComponent, ContainerIsInsertingAttemptEvent>(OnInsertIntoContainer);
+        SubscribeLocalEvent<ItemSlotRendererComponent, ContainerIsRemovingAttemptEvent>(OnRemoveFromContainer);
+
+    }
+    private void OnInsertIntoContainer(EntityUid uid, ItemSlotRendererComponent comp, ContainerIsInsertingAttemptEvent args)
+    {
+        if (args.Container is not ContainerSlot || !_timing.IsFirstTimePredicted)
+            return;
+
+        comp.CachedEntities[args.Container.ID] = args.EntityUid;
+    }
+
+    private void OnRemoveFromContainer(EntityUid uid, ItemSlotRendererComponent comp, ContainerIsRemovingAttemptEvent args)
```

Comment:
this event is fired by `CanRemove()`, which is not good
this should've used `EntRemovedFromContainerMessage` instead